lib/types/marks.dart

```diff
@@ -47,17 +47,23 @@ class MARKS {
     MARKS.UNDERLINE.value: TextStyle(decoration: TextDecoration.underline),
   });
 
-  static TextStyle getMarksTextStyles(
-      List<Mark> marks, Map<dynamic, TextStyle> renderMark) {
-    Map<String, TextStyle?> textStyles = {};
+  static TextStyle getMarksTextStyles(List<Mark> marks, Map<dynamic, TextStyle> renderMark) {
+    TextStyle combinedStyle = TextStyle();
+    TextStyle combinedTypeStyle = TextStyle();
+
     marks.forEach((Mark mark) {
-      textStyles.putIfAbsent(mark.type, () => renderMark[mark.type]);
+      if (renderMark.containsKey(mark.type)) {
+        TextStyle markStyle = renderMark[mark.type]!;
+        combinedStyle = combinedStyle.merge(markStyle);
+        combinedTypeStyle = combinedTypeStyle.copyWith(
+          fontWeight: mark.type == 'bold' ? markStyle.fontWeight : combinedTypeStyle.fontWeight,
+          fontStyle: mark.type == 'italic' ? markStyle.fontStyle : combinedTypeStyle.fontStyle,
+          decoration: mark.type == 'underline' ? markStyle.decoration : combinedTypeStyle.decoration,
+        );
+      }
     });
-    return TextStyle(
-      fontWeight: textStyles['bold']?.fontWeight,
-      fontStyle: textStyles['italic']?.fontStyle,
-      decoration: textStyles['underline']?.decoration,
-    );
+
+    return combinedTypeStyle.merge(combinedStyle);
```

Comment:
If I understand the code correctly, this should have been switched. I guess you want to overwrite the style that contains all values from all marks (`combinedStyle`) with the specific values for `fontWeight`, `fontStyle` and `decoration` coming from the `markStyle`s. To make sure that for example the mark style for "underline" cannot overwrite the `FontStyle.italic` from the "italic" mark style.
```suggestion
    return combinedStyle.merge(combinedTypeStyle);
```

Reply:
Good point @lColinDl 👍